portal/server/controllers/utilisation-report-service/utilisation-report-upload/utilisation-report-validator.js

```diff
@@ -0,0 +1,140 @@
+// Errors can look like:
+const {
+  generateUkefFacilityIdError,
+  generateBaseCurrencyError,
+  generateFacilityUtilisationError,
+  generateMonthlyFeesPaidError,
+  generateTotalFeesAccruedError,
+  generatePaymentCurrencyError,
+  generateExchangeRateError,
+} = require('./utilisation-report-cell-validators');
+const { HEADERS } = require('../../../constants');
+
+const validateCsvHeaders = (csvDataRow) => {
+  const headers = Object.keys(csvDataRow);
+  const missingHeaderErrors = [];
+  const availableHeaders = [];
+
+  if (!headers.includes(HEADERS.UKEF_FACILITY_ID)) {
+    missingHeaderErrors.push({
+      errorMessage: 'UKEF facility ID header is missing or spelt incorrectly',
+      column: undefined,
+      row: null,
+      value: null,
+      exporter: null,
+    });
+  } else {
+    availableHeaders.push(HEADERS.UKEF_FACILITY_ID);
+  }
+
+  if (!headers.includes(HEADERS.BASE_CURRENCY)) {
+    missingHeaderErrors.push({ errorMessage: 'Base currency header is missing or spelt incorrectly', column: null, row: null, value: null, exporter: null });
+  } else {
+    availableHeaders.push(HEADERS.BASE_CURRENCY);
+  }
+
+  if (!headers.includes(HEADERS.FACILITY_UTILISATION)) {
+    missingHeaderErrors.push({
+      errorMessage: 'Facility utilisation header is missing or spelt incorrectly',
+      column: null,
+      row: null,
+      value: null,
+      exporter: null,
+    });
+  } else {
+    availableHeaders.push(HEADERS.FACILITY_UTILISATION);
+  }
+
+  if (!headers.includes(HEADERS.TOTAL_FEES_ACCRUED)) {
+    missingHeaderErrors.push({
+      errorMessage: 'Total fees accrued for the month header is missing or spelt incorrectly',
+      column: null,
+      row: null,
+      value: null,
+      exporter: null,
+    });
+  } else {
+    availableHeaders.push(HEADERS.TOTAL_FEES_ACCRUED);
+  }
+
+  if (!headers.includes(HEADERS.MONTHLY_FEES_PAID)) {
+    missingHeaderErrors.push({
+      errorMessage: 'Monthly fees paid to UKEF header is missing or spelt incorrectly',
+      column: null,
+      row: null,
+      value: null,
+      exporter: null,
+    });
+  } else {
+    availableHeaders.push(HEADERS.MONTHLY_FEES_PAID);
+  }
+  return { missingHeaderErrors, availableHeaders };
+};
+
+const validateCsvCellData = (csvData, availableHeaders) => csvData.flatMap((value, index) => {
```

Comment:
This function would be good to reduce duplication in as well, maybe we can do something similar to the above comment